compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUVectorization.cpp

```diff
@@ -107,10 +110,78 @@ static SmallVector<int64_t> getCanonicalVectorShape(func::FuncOp funcOp) {
   return canonicalVectorShape;
 }
 
+/// Tries to infer the vector sizes from an IR that has been tiled for
+/// vectorization. Returns failure if vector sizes can't be inferred. For know
+/// this utility is only able to extract tile size information from
+/// `tensor.extract_slice` and `affine.min` ops, which should cover most but not
+/// all the cases.
+static FailureOr<SmallVector<int64_t>> inferVectorSizesFromIR(
+    linalg::LinalgOp linalgOp) {
+  LLVM_DEBUG(VEC_DBGS() << "Inferring vector sizes for:\n" << linalgOp << "\n");
+
+  auto idxMaps = linalgOp.getIndexingMapsArray();
+  assert(!idxMaps.empty());
+  unsigned numDims = idxMaps[0].getNumDims();
+
+  SmallVector<int64_t> vectorSizes;
+  for (int dim = 0; dim < numDims; ++dim) {
+    // Map dimension `dim` to an operand dimension that we will use to
+    // traverse the U-D chain to get `dim` vector size information.
+    SmallVector<std::pair<Value, unsigned>> operandDimPairs;
+    linalgOp.mapIterationSpaceDimToAllOperandDims(dim, operandDimPairs);
+    if (operandDimPairs.empty()) {
+      return failure();
+    }
+
+    Value firstOperand = operandDimPairs[0].first;
+    unsigned firstOperandDim = operandDimPairs[0].second;
+
+    // Trivial case: `dim` size is available in the operand type.
+    int64_t dimSize =
+        firstOperand.getType().cast<ShapedType>().getShape()[firstOperandDim];
+    if (!ShapedType::isDynamic(dimSize)) {
+      vectorSizes.push_back(dimSize);
+      continue;
+    }
```

Comment:
IMO, we can add a debug log for the trivial case as well. I'd be confused to not seeing the log if I enable the `debug-only`.